### ISSUE TITLE
Regarding other types of exercises

### ISSUE THREAD

**Issue by geniusupgrader** [brodeurlv/fastnfitness#33]
For now we have two categories:

- Body Building
- Cardio (Beta)

The Body Building exercises are described with: "Repetitions" and "Weight". (And multiple ones can concatenated with "Series".)
The Cardio exercises are described with: "Distance" and "Duration".

But there are other exercises, which require different types of descriptions, like:

- Only Distance
- Only Duration
- Only Weight
- Only Reps
- Weight and Distance
- Weight and Duration
- Reps and distance
- Reps and Duration
- (And of course all other combinations)

The App [FitNotes](https://play.google.com/store/apps/details?id=com.github.jamesgay.fitnotes) defines these types at every exercise (on the exercise level) and not for the whole category as it is done in FastnFitness. I don't know which method is better, but the fact that the Cardio exercises are not visible in the "Exercises" tab/page is not very nice. The Exercises page should contain all exercises, or should'n it?

Personally, I really miss the type "Reps and Duration" for doing callisthenics exercises, which often are static hold exercises. For that I would really like to help, but this needs discussion, because there are multiple ways to implement:

1. Add a new category: **callisthenics** (beside Body Building and Cardio (Beta)), where every exercise has "Reps and Duration" as its type.
2. Combine the two categories: "Body Building and Cardio (Beta)" to one (Maybe: "Workout"?). Then every exercise has its own type as in the list from above, which automatically opens when selecting an exercise.

Number 1. has the problem, that you would propably need even more categories in the future. Also this would (under certain circumstances) duplicate code multiple times.
I favor Number 2. Because for now the Cardio is still in Beta and the app is not this big to incorporate such a change. Maybe Number 2 sounds like it is way more complicated, but I guess it would even simplify the UI.

Other opinions?

**Comment by geniusupgrader** [comment]
Almost forgotten: a tempo description can be added for every exercise (how fast you do the exercise)

**Comment by brodeurlv** [comment]
Hi, 

I'm currently merging the 2 fragments Cardio and Fontes. It will be much easier in term of maintenance and evolution. This will be an enabler to add other kind of records. 

Thanks for the detailled comments here. It is very helpful.

**Comment by geniusupgrader** [comment]
Thanks for tackling this.

**Comment by geniusupgrader** [comment]
I want to empahasize this:
**FastnFitness should become more general** in its exercises and everything (not only for Weight lifting).

**Comment by brodeurlv** [comment]
yes you're right. I'm progressing on the development but the impact it significant. I need to make sure there is no regression.  You can have a look at my MergedUI branch if you want to see in which direction I'm heading.
Don't hesitate if you have remarks.

**Comment by brodeurlv** [comment]
Both cardio and bodybuilding UI are merged in 0.16.
I have not addressed other types of exercises but I'm thinking about it for after.

**Comment by geniusupgrader** [comment]
thank, what about pull request  #46 from me. Is it compatible?

**Comment by vanous** [comment]
Hi all, is there any plan for progress on this one? I now add stuff like Pilates as cardio 1km 45minutes, but am not sure what will it do in the future. Thank you.

**Comment by ovz93br43v7** [comment]
I also would like to see support for "free exercises" like reps rope climbing in x minutes.
And AMRAP work-outs (as many rounds as possible), which means combination of different exercises as fast as you can within a time cap.

**Comment by geniusupgrader** [comment]
Maybe AMRAP workouts are a bit out of scope for now...
